### DISCUSSION TITLE
helm: How to configure storage backends?

### DISCUSSION THREAD

**Discussion by aviau** [seaweedfs/seaweedfs#4796] [Q&A]
Hello,

We are trying to configure storage backends using the helm chart.

I'd like to setup S3 as a storage backend as documented in the [Cloud Tier page](https://github.com/seaweedfs/seaweedfs/wiki/Cloud-Tier#s3) in the wiki. The documentation suggests a configuration that looks like this:
```
[storage.backend]
	[storage.backend.s3.default]
	enabled = true
        ...
```

Since it does not seem possible to modify `master.toml` from the helm chart, I assume that the intended way to perform this is using environment variables.

I have tried the following helm `values`:
```
master:
  extraEnvironmentVars:
    WEED_STORAGE_BACKEND_S3_DEFAULT_ENABLED: "true"
```

Now, I am not sure where to check if I want to confirm that it worked or not. I tried using `weed shell` and perform the commands listed in the wiki page:
```
> lock
> volume.tier.upload -dest s3.default -volumeId=1
markVolumeReadonly 1 on seaweedfs-volume-0.seaweedfs-volume.default:8080 ...
error: copy dat file for volume 1 on seaweedfs-volume-0.seaweedfs-volume.default:8080 to s3.default: rpc error: code = Unknown desc = destination s3.default not found, supported: []
> volume.tier.upload -dest s3 -volumeId=1
markVolumeReadonly 1 on seaweedfs-volume-0.seaweedfs-volume.default:8080 ...
error: copy dat file for volume 1 on seaweedfs-volume-0.seaweedfs-volume.default:8080 to s3: rpc error: code = Unknown desc = destination s3 not found, supported: []
```

It looks like the `s3` or `s3.default` destinations are not found, what am I doing wrong?

**Comment by chrislusf** [comment]
There is an additional string map for `storage.backend`, which may have bugs to handle mapping env variables to configuration variables.
 https://github.com/seaweedfs/seaweedfs/blob/03389bf8441d6130bc36b4a1732e5dc12fb8fd2e/weed/storage/backend/backend.go#L52
